awx/main/utils/db.py

```diff
@@ -4,7 +4,22 @@
 
 from awx.settings.application_name import set_application_name
 from django.conf import settings
+from django.db import transaction
 
 
 def set_connection_name(function):
     set_application_name(settings.DATABASES, settings.CLUSTER_HOST_ID, function=function)
+
+
+def bulk_update_sorted_by_id(model, objects, fields, batch_size=1000):
+    # Filter out objects with None ID
+    objects = [obj for obj in objects if obj.id is not None]
+    # If there are no valid objects, return early
+    if not objects:
+        return
+    # Sort objects by their ID
+    sorted_objects = sorted(objects, key=lambda obj: obj.id)
+
+    # Perform the bulk update within an atomic transaction
+    with transaction.atomic():
```

Comment:
The bulk update is a single statement, a transaction should not be necessary at all. The task system should usually, as a best practice, operate in auto-commit mode. If you don't, the read-write flow behaves in a way that's not intuitive and makes cross-process errors more likely due to uncommited changes in another client's transaction.

So I think the best course of action is to just not do this in a transaction.